src/main/scala/Dromajo.scala

```diff
@@ -0,0 +1,101 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.{IntParam, StringParam}
+import freechips.rocketchip.config.{Parameters}
+
+/**
+ * Dromajo bridge to input instruction streams and check with Dromajo
+ */
+class SimDromajoBridge(insnWidths: TracedInstructionWidths, numInsns: Int) extends Module
+{
+  val io = IO(new Bundle {
+    val trace = Input(new TileTraceIO(insnWidths, numInsns))
+  })
+
+  // constants
+  val xLen = 64
+  val instBits = 32
+  // constants
+
+  val traces = io.trace.insns
+
+  val dromajo = Module(new SimDromajoCosimBlackBox(numInsns, xLen))
+
+  dromajo.io.clock := clock
+  dromajo.io.reset := reset.asBool
+
+  // TODO: convert to use SInts?
+  def signextPad(in: UInt, padLen: Int): UInt = {
```

Comment:
Seems like you'll need to keep using a function:
https://github.com/chipsalliance/rocket-chip/blob/6ac71323666eee369e326c1dc4e8036a869cea28/src/main/scala/util/package.scala#L104

Reply:
Ill probably switch to this other function then. Good catch.